src/metadata/metadata.controller.spec.ts

```diff
@@ -0,0 +1,169 @@
+import { createMockTransactionResult, MockMetadataEntry } from '~/test-utils/mocks';
+import { mockMetadataServiceProvider } from '~/test-utils/service-mocks';
+
+describe('MetadataController', () => {
+  let controller: MetadataController;
+  let mockService: DeepMocked<MetadataService>;
+  let ticker: string;
+  let type: MetadataType;
+  let id: BigNumber;
+
+  beforeEach(async () => {
+    ticker = 'TICKER';
+    type = MetadataType.Local;
+    id = new BigNumber(1);
+
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [MetadataController],
+      providers: [mockMetadataServiceProvider],
+    }).compile();
+
+    mockService = mockMetadataServiceProvider.useValue as DeepMocked<MetadataService>;
+
+    controller = module.get<MetadataController>(MetadataController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
+  describe('getMetadata', () => {
+    it('should return the list of all metadata for a given ticker', async () => {
+      const mockMetadataEntry = new MockMetadataEntry();
+      when(mockService.findAll)
+        .calledWith(ticker)
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .mockResolvedValue([mockMetadataEntry as any]);
+
+      const result = await controller.getMetadata({ ticker });
+
+      expect(result).toEqual({
+        results: [new MetadataEntryModel({ asset: ticker, type, id })],
+      });
+    });
+  });
+
+  describe('getSingleMetadata', () => {
+    it('should return the Metadata details for a specific type and ID', async () => {
+      const mockMetadataEntry = new MockMetadataEntry();
+      const mockDetails = {
+        name: 'Some metadata',
+        specs: {
+          description: 'Some description',
+        },
+      };
+
+      const mockValue = {
+        value: 'Some Value',
+        expiry: new Date('2099/01/01'),
+        lockStatus: MetadataLockStatus.LockedUntil,
+        lockedUntil: new Date('2030/01/01'),
+      };
+
+      mockMetadataEntry.details.mockResolvedValue(mockDetails);
+      mockMetadataEntry.value.mockResolvedValue(mockValue);
+
+      when(mockService.findOne)
+        .calledWith({ ticker, type, id })
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .mockResolvedValue(mockMetadataEntry as any);
+
+      const result = await controller.getSingleMetadata({ ticker, type, id });
+
+      expect(result).toEqual(
+        new MetadataDetailsModel({
+          asset: ticker,
+          type,
+          id,
+          ...mockDetails,
+          value: new MetadataValueModel(mockValue),
+        })
+      );
+    });
+  });
+
+  describe('createMetadata', () => {
+    it('should accept CreateMetadataDto and create a local Asset Metadata for the given ticker', async () => {
+      const transaction = {
+        blockHash: '0x1',
+        transactionHash: '0x2',
+        blockNumber: new BigNumber(1),
```

Comment:
(optional) Just an idea, but we could put these into `src/test-utils/consts.ts` and call it something like `txBase` and do something like this:

```suggestion
        ...txBase
```

OK to leave as it though

Reply:
After Toms change, we can refactor couples of files to match this pattern. For now I'll leave like this